### PR TITLE
chore: deprecate SearcherContext

### PR DESCRIPTION
**Pull request by rb-determined-ai**
Instead of asking the master for SearcherOperations, the SearcherContext
fabricates a single SearcherOperation by looking at the max_length of
the experiment config.

Both the max_length of the experiment config and the SearcherContext
itself are deprecated.  If the max_length is not present, the
SearcherContext will throw an error if it is used.

### CODE REVIEW

File: docs/model-dev-guide/api-guides/apis-howto/api-core-ug-basic.rst
Comment (tara-det-ai) [inline docs/model-dev-guide/api-guides/apis-howto/api-core-ug-basic.rst:1]: Do we need to call it out literally, e.g., "The SearcherContext fabricates a single SearcherOperation based on the ``max_length`` specified in the experiment configuration. Ensure that your configuration includes a ``max_length`` parameter in the searcher section."

File: docs/model-dev-guide/api-guides/apis-howto/api-core-ug.rst [hunk 1]
```diff
@@ -334,6 +334,7 @@ settings in our experiment configuration file:
 -  ``max_trials``: 500 (This is the maximum number of trials the searcher should run.)
 -  ``max_length``: 20 epochs (The max length of a trial. For more information, visit Adaptive ASHA
```
Comment (tara-det-ai) [inline docs/model-dev-guide/api-guides/apis-howto/api-core-ug.rst:335]: -  ``max_length``: 20 epochs (The max length of a trial.  This is a required parameter for the SearcherContext to function properly.)

File: docs/model-dev-guide/api-guides/apis-howto/api-core-ug-basic.rst
Comment (rb-determined-ai) [reply docs/model-dev-guide/api-guides/apis-howto/api-core-ug-basic.rst:1]: No, my goal is to strip all knowledge of SearcherContext and max_length from the tutorial.  You don't need it, and you don't need to know how we fake it for backwards compatibility.

**Comment by rb-determined-ai** [comment]
pushed this and more directly to the `searcher-context-removal` branch, closing now.

**Pull request closed without merging**